### PR TITLE
SAML: update docs

### PR DESCRIPTION
**Pull request by stsewd**
- Re-arrange and rename some sections, so the user follows a more logical order.
- Adding detailed steps for enabling SAML on Read the Docs, leaving just the domain configuration to do it via support.
- Clarify user management and team configuration

preview at https://docs--11583.org.readthedocs.build/en/11583/guides/set-up-single-sign-on-saml.html

---
:books: Documentation previews :books:

- User's documentation (`docs`): https://docs--11583.org.readthedocs.build/en/11583/

- Developer's documentation (`dev`): https://dev--11583.org.readthedocs.build/en/11583/

### CODE REVIEW

File: docs/user/guides/set-up-single-sign-on-saml.rst [hunk 1]
```diff
@@ -51,56 +51,78 @@ In order to enable SSO with Okta, you need to create a new SAML application in y
...
-------------------------------------
-
-You can mark one or more teams that users will automatically join when they sign up with a matching email address.
-Configure this option by:
+To configure a team for users to join automatically:
 
 1. Navigate to the `teams management page <https://readthedocs.com/organizations/choose/organization_team_list/>`__.
 2. Click the :guilabel:`<team name>`.
 3. Click :guilabel:`Edit team`
 4. Enable *Auto join users with an organization's email address to this team*.
```
Comment (ericholscher) [inline docs/user/guides/set-up-single-sign-on-saml.rst:87]: Seems like this copy should be updated to mention SAML when that applies?

File: docs/user/guides/set-up-single-sign-on-saml.rst [hunk 2]
```diff
@@ -51,56 +51,78 @@ In order to enable SSO with Okta, you need to create a new SAML application in y
...
+~~~~~~~~~
+
+After enabling the SAML integration,
+all users with an email domain matching the one in your SAML integration will be required to sign up using SAML.
+After they sign up, they will be automatically joined to your organization within the teams you have configured to auto-join users.
+
+Existing users
+~~~~~~~~~~~~~~
+
+Existing users with email addresses from your configured domain will not be required to sign up using SAML,
```
Comment (ericholscher) [inline docs/user/guides/set-up-single-sign-on-saml.rst:103]: Do we mean sign in here? 
```suggestion
Existing users with email addresses from your configured domain will not be required to sign in using SAML,
```

File: docs/user/guides/set-up-single-sign-on-saml.rst [hunk 3]
```diff
@@ -51,56 +51,78 @@ In order to enable SSO with Okta, you need to create a new SAML application in y
...
+but they won't be automatically joined to your organization.
+
+If you want to enforce SAML for existing users, you have the following options:
+
+- Users can delete their accounts, and sign up again using SAML.
+- Users can link their existing accounts to their SAML identity by following this link while logged in their Read the Docs account:
+  ``https://readthedocs.com/accounts/saml/<organization-slug>/login/?process=connect`` (replace ``<organization-slug>`` with your organization slug).
```
Comment (ericholscher) [inline docs/user/guides/set-up-single-sign-on-saml.rst:110]: Is this link exposed to users anywhere? We should expose it, and explain users where to get that link here./

**Review by ericholscher — APPROVED** [review]

File: docs/user/guides/set-up-single-sign-on-saml.rst (hunk 3 above)
Comment (stsewd) [reply docs/user/guides/set-up-single-sign-on-saml.rst:110]: We do show this in the SAML settings page. Added a link to that page.

File: docs/user/guides/set-up-single-sign-on-saml.rst (hunk 1 above)
Comment (stsewd) [reply docs/user/guides/set-up-single-sign-on-saml.rst:87]: Hmm, maybe directly mention the domain?

`Automatically join users that signup with SAML using a @example.com email address`

And for google

`Automatically join users that signup with Google using a @example.com email address`

Comment (stsewd) [reply docs/user/guides/set-up-single-sign-on-saml.rst:87]: But I'm also fine keeping the copy generic, maybe improve it a little

> Automatically add users who signup with your organization's email address to this team

Comment (stsewd) [reply docs/user/guides/set-up-single-sign-on-saml.rst:87]: https://github.com/readthedocs/readthedocs-corporate/issues/1875

**Pull request merged**